// nav-frontend-spinner/src/spinner.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const PT = require("prop-types");
const React = require("react");
const classNames = require("classnames");
const nav_frontend_ikoner_assets_1 = require("nav-frontend-ikoner-assets");
require("nav-frontend-spinner-style");
const cls = (className, storrelse) => classNames('spinner', className, {
    [`spinner--${storrelse.toLowerCase()}`]: !!storrelse
});
class NavFrontendSpinner extends React.Component {
    render() {
        const { negativ, stroke, type, className, ...props } = this.props;
        const ikon = stroke ? 'spinner-stroke' : 'spinner';
        const ikonNegativ = negativ ? `${ikon}-negativ` : ikon;
        return (React.createElement(nav_frontend_ikoner_assets_1.default, Object.assign({ kind: ikonNegativ, className: cls(className, type) }, props)));
    }
}
NavFrontendSpinner.propTypes = {
    negativ: PT.bool,
    stroke: PT.bool,
    type: PT.oneOf(['XXS', 'XS', 'S', 'M', 'L', 'XL', 'XXL', 'XXXL']),
    className: PT.string,
    'aria-label': PT.string
};
NavFrontendSpinner.defaultProps = {
    'aria-label': 'Laster innhold',
    negativ: false,
    stroke: false,
    type: 'M',
    className: undefined
};
exports.default = NavFrontendSpinner;
