// nav-frontend-etiketter/src/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const PT = require("prop-types");
const React = require("react");
const classNames = require("classnames");
const nav_frontend_typografi_1 = require("nav-frontend-typografi");
require("nav-frontend-etiketter-style");
const cls = (type, className) => classNames('etikett', className, {
    'etikett--advarsel': type === 'advarsel',
    'etikett--fokus': type === 'fokus',
    'etikett--suksess': type === 'suksess',
    'etikett--info': type === 'info'
});
class EtikettBase extends React.Component {
    render() {
        const { type, className, typo = 'normaltekst', children, ...props } = this.props;
        return (React.createElement("div", Object.assign({ className: cls(type, className) }, props),
            "Bingo:",
            React.createElement(nav_frontend_typografi_1.default, { tag: "span", type: typo }, children)));
    }
}
EtikettBase.propTypes = {
    /**
     * Typografi som skal brukes i etiketten. Støtter samme typer som `TypografiBase`. Påvirker ikke tag...
     */
    typo: PT.string,
    children: PT.node.isRequired,
    /**
     * Predefinerte farger
     */
    type: PT.oneOf(['suksess', 'info', 'fokus', 'advarsel']).isRequired,
    className: PT.string
};
EtikettBase.defaultProps = {
    typo: 'normaltekst',
    className: undefined
};
exports.default = EtikettBase;
var etikettadvarsel_1 = require("./etikettadvarsel");
exports.EtikettAdvarsel = etikettadvarsel_1.default;
var etikettfokus_1 = require("./etikettfokus");
exports.EtikettFokus = etikettfokus_1.default;
var etikettsuksess_1 = require("./etikettsuksess");
exports.EtikettSuksess = etikettsuksess_1.default;
var etikettinfo_1 = require("./etikettinfo");
exports.EtikettInfo = etikettinfo_1.default;
