// nav-frontend-skjema/src/checkbox.tsx
import PT from 'prop-types';
/* eslint-disable react/prefer-stateless-function  */
import * as React from 'react';
import * as classNames from 'classnames';
import { guid } from 'nav-frontend-js-utils';
import 'nav-frontend-skjema-style'; // eslint-disable-line import/extensions
import SkjemaelementFeilmelding, {
    skjemaelementFeilmeldingShape,
    SkjemaelementFeilmeldingType
} from './skjemaelement-feilmelding';

const cls = (className) => classNames('skjemaelement skjemaelement--horisontal', className);
const inputCls = (harFeil) => classNames('skjemaelement__input checkboks', {
    'skjemaelement__input--harFeil': harFeil
});

export interface CheckboxProps {
    label: string;
    className?: string;
    id?:string;
    feil?: SkjemaelementFeilmeldingType;
    checkboxRef?: () => void
}

/**
 * Checkbox
 */
class Checkbox extends React.Component<CheckboxProps> {

    render() {
        const { className, label, id, feil, checkboxRef, ...other } = this.props;
        const inputId = id || guid();
        return (
            <div className={cls(className)}>
                <input
                    type="checkbox"
                    className={inputCls(feil)}
                    id={inputId}
                    ref={checkboxRef}
                    {...other}
                />
                <label className="skjemaelement__label" htmlFor={inputId}>{label}</label>
                <SkjemaelementFeilmelding feil={feil} />
            </div>
        );
    }
}

(Checkbox as React.ComponentClass).propTypes = {
    /**
     * ClassName
     */
    className: PT.string,
    /**
     * Label for checkbox
     */
    label: PT.node.isRequired,
    /**
     * Id for checkbox, hvis id ikke er satt brukes en tilfeldig guid
     */
    id: PT.string,
    /**
     * Hvis skjemaet har feil sender man inn et objekt med en feilmelding
     */
    feil: skjemaelementFeilmeldingShape,
    /**
     * Referanse til selve checkboxen. Brukes for eksempel til å sette fokus
     */
    checkboxRef: PT.func
};

(Checkbox as React.ComponentClass).defaultProps = {
    className: undefined,
    id: undefined,
    feil: undefined,
    checkboxRef: undefined
};

export default Checkbox;
