// nav-frontend-skjema/src/checkbox.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const PT = require("prop-types");
const React = require("react");
const nav_frontend_js_utils_1 = require("nav-frontend-js-utils");
require("nav-frontend-skjema-style");
const classNames = require("classnames");
const skjemaelement_feilmelding_1 = require("./skjemaelement-feilmelding"); // tslint:disable-line:max-line-length
const cls = (className) => classNames('skjemaelement skjemaelement--horisontal', className);
const inputCls = (harFeil) => classNames('skjemaelement__input checkboks', {
    'skjemaelement__input--harFeil': harFeil
});
/**
 * Checkbox
 */
class Checkbox extends React.Component {
    render() {
        const { className, label, id, feil, checkboxRef, ...other } = this.props;
        const inputId = id || nav_frontend_js_utils_1.guid();
        // tslint:disable:react-a11y-role-has-required-aria-props
        return (React.createElement("div", { className: cls(className) },
            React.createElement("input", Object.assign({ type: "checkbox", className: inputCls(feil), id: inputId, ref: checkboxRef }, other)),
            React.createElement("label", { className: "skjemaelement__label", htmlFor: inputId }, label),
            React.createElement(skjemaelement_feilmelding_1.default, { feil: feil })));
    }
}
exports.Checkbox = Checkbox;
Checkbox.propTypes = {
    /**
     * ClassName
     */
    className: PT.string,
    /**
     * Label for checkbox
     */
    label: PT.node.isRequired,
    /**
     * Id for checkbox, hvis id ikke er satt brukes en tilfeldig guid
     */
    id: PT.string,
    /**
     * Hvis skjemaet har feil sender man inn et objekt med en feilmelding
     */
    feil: skjemaelement_feilmelding_1.skjemaelementFeilmeldingShape,
    /**
     * Referanse til selve checkboxen. Brukes for eksempel til å sette fokus
     */
    checkboxRef: PT.func
};
Checkbox.defaultProps = {
    className: undefined,
    id: undefined,
    feil: undefined,
    checkboxRef: undefined
};
exports.default = Checkbox;
