// nav-frontend-skjema/src/checkboks-panel-gruppe.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
const PT = require("prop-types");
require("nav-frontend-skjema-style");
const _1 = require("./");
const skjemaelement_feilmelding_1 = require("./skjemaelement-feilmelding");
const checkboks_panel_1 = require("./checkboks-panel");
class CheckboksPanelGruppe extends React.Component {
    render() {
        const { checkboxes, legend, feil, onChange } = this.props;
        return (React.createElement(_1.SkjemaGruppe, { className: "inputPanelGruppe", feil: feil },
            React.createElement(_1.Fieldset, { legend: legend }, checkboxes.map((checkbox) => (React.createElement(checkboks_panel_1.CheckboksPanel, Object.assign({ key: `${checkbox.id}${checkbox.label}`, onChange: (event) => onChange(event, checkbox.value) }, checkbox)))))));
    }
}
CheckboksPanelGruppe.propTypes = {
    checkboxes: PT.array.isRequired,
    legend: PT.string.isRequired,
    onChange: PT.func.isRequired,
    feil: skjemaelement_feilmelding_1.skjemaelementFeilmeldingShape
};
CheckboksPanelGruppe.defaultProps = {
    feil: undefined
};
exports.default = CheckboksPanelGruppe;
