// nav-frontend-skjema/src/fieldset.tsx
/* eslint-disable react/prefer-stateless-function  */
import * as PT from 'prop-types';
import * as React from 'react';
import * as classNames from 'classnames';
import 'nav-frontend-skjema-style';

const cls = (className) => classNames('skjema__fieldset', className);

export interface FieldsetProps {
    children: React.ReactChildren | React.ReactChild | React.ReactNode;
    legend: string;
    className?: string;
}

/**
 * Fieldset
 */
class Fieldset extends React.Component<FieldsetProps> {
    render() {
        const { children, className, legend, ...other } = this.props;
        return (
            <fieldset className={cls(className)} {...other}>
                <legend className="skjema__legend">{legend}</legend>
                {children}
            </fieldset>
        );
    }
}

(Fieldset as React.ComponentClass).propTypes = {
    /**
     * En eller flere children, oftes .skjemaelement__sporsmal eller .skjemaelement
     */
    children: PT.oneOfType([
        PT.arrayOf(PT.node),
        PT.node
    ]).isRequired,
    /**
     * ClassName
     */
    className: PT.string,
    /**
     * legend for fieldset
     */
    legend: PT.string.isRequired
};

(Fieldset as React.ComponentClass).defaultProps = {
    className: undefined
};

export default Fieldset;
