// nav-frontend-skjema/src/input.tsx
import * as classNames from 'classnames';
import * as React from 'react';
import PT from 'prop-types';
import { guid } from 'nav-frontend-js-utils';
import 'nav-frontend-skjema-style'; // eslint-disable-line import/extensions
import SkjemaelementFeilmelding, {
    skjemaelementFeilmeldingShape,
    SkjemaelementFeilmeldingType
} from './skjemaelement-feilmelding';
import Bredde from './index'

const cls = (className) => classNames('skjemaelement', className);
const inputClass = (width, className, harFeil) => classNames(
    'skjemaelement__input',
    className,
    `input--${width.toLowerCase()}`,
    { 'skjemaelement__input--harFeil': harFeil }
);

export interface InputProps {
    label: React.ReactNode;
    feil?: SkjemaelementFeilmeldingType;
    bredde?: Bredde;
    id?: string;
    name?: string;
    inputRef?: () => void;
    className?: string;
    inputClassName?: string;
}


/**
 * Standard inputfelt med mulighet for 책 sette bredde
 */
class Input extends React.Component<InputProps> { // eslint-disable-line react/prefer-stateless-function
    render() {
        const { label, bredde, feil, id, name, inputRef, className, inputClassName, ...other } = this.props;
        const inputId = id || name || guid();

        return (
            <div className={cls(className)}>
                <label className="skjemaelement__label" htmlFor={inputId}>{label}</label>
                <input
                    type="text"
                    className={inputClass(bredde, inputClassName, feil)}
                    id={inputId}
                    {...other}
                    ref={inputRef}
                />
                <SkjemaelementFeilmelding feil={feil} />
            </div>
        );
    }
}

(Input as React.ComponentClass).propTypes = {
    /**
     * Label for tekstfeltet
     */
    label: PT.node.isRequired,
    /**
     * Hvis skjemaet har feil sender man inn et objekt med en feilmelding
     */
    feil: skjemaelementFeilmeldingShape,
    /**
     * Bredden til inputfeltet
     */
    bredde: PT.oneOf(['fullbredde', 'XXL', 'XL', 'L', 'M', 'S', 'XS', 'XXS']),
    /**
     * Id for inputfelt, hvis id ikke er satt brukes name eller en tilfeldig guid
     */
    id: PT.string,
    /**
     * Name for inputfelt, hvis name ikke er satt brukes id eller en tilfeldig guid
     */
    name: PT.string,
    /**
     * Referanse til selve inputfeltet. Brukes for eksempel til 책 sette fokus
     */
    inputRef: PT.func,
    /**
     * Classname som blir satt p책 komponentwrapperen
     */
    className: PT.string,
    /**
     * Classname som blir satt p책 input-elementet
     */
    inputClassName: PT.string
};

(Input as React.ComponentClass).defaultProps = {
    bredde: 'fullbredde',
    feil: undefined,
    id: undefined,
    name: undefined,
    inputRef: undefined,
    className: undefined,
    inputClassName: undefined
};

export default Input;
