// nav-frontend-skjema/src/input.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
const PT = require("prop-types");
const nav_frontend_js_utils_1 = require("nav-frontend-js-utils");
require("nav-frontend-skjema-style");
const classNames = require("classnames");
const skjemaelement_feilmelding_1 = require("./skjemaelement-feilmelding"); // tslint:disable-line:max-line-length
const cls = (className) => classNames('skjemaelement', className);
const inputClass = (width, className, harFeil) => classNames('skjemaelement__input', className, `input--${width.toLowerCase()}`, { 'skjemaelement__input--harFeil': harFeil });
/**
 * Standard inputfelt med mulighet for 책 sette bredde
 */
class Input extends React.Component {
    render() {
        const { label, bredde, feil, id, name, inputRef, className, inputClassName, ...other } = this.props;
        const inputId = id || name || nav_frontend_js_utils_1.guid();
        return (React.createElement("div", { className: cls(className) },
            React.createElement("label", { className: "skjemaelement__label", htmlFor: inputId }, label),
            React.createElement("input", Object.assign({ type: "text", className: inputClass(bredde, inputClassName, feil), id: inputId, name: name }, other, { ref: inputRef })),
            React.createElement(skjemaelement_feilmelding_1.default, { feil: feil })));
    }
}
Input.propTypes = {
    /**
     * Label for tekstfeltet
     */
    label: PT.node.isRequired,
    /**
     * Hvis skjemaet har feil sender man inn et objekt med en feilmelding
     */
    feil: skjemaelement_feilmelding_1.skjemaelementFeilmeldingShape,
    /**
     * Bredden til inputfeltet
     */
    bredde: PT.oneOf(['fullbredde', 'XXL', 'XL', 'L', 'M', 'S', 'XS', 'XXS']),
    /**
     * Id for inputfelt, hvis id ikke er satt brukes name eller en tilfeldig guid
     */
    id: PT.string,
    /**
     * Name for inputfelt, hvis name ikke er satt brukes id eller en tilfeldig guid
     */
    name: PT.string,
    /**
     * Referanse til selve inputfeltet. Brukes for eksempel til 책 sette fokus
     */
    inputRef: PT.func,
    /**
     * Classname som blir satt p책 komponentwrapperen
     */
    className: PT.string,
    /**
     * Classname som blir satt p책 input-elementet
     */
    inputClassName: PT.string
};
Input.defaultProps = {
    bredde: 'fullbredde',
    feil: undefined,
    id: undefined,
    name: undefined,
    inputRef: undefined,
    className: undefined,
    inputClassName: undefined
};
exports.default = Input;
