// nav-frontend-skjema/src/skjema-gruppe.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const PT = require("prop-types");
const React = require("react");
const classNames = require("classnames");
require("nav-frontend-skjema-style");
const skjemaelement_feilmelding_1 = require("./skjemaelement-feilmelding"); // tslint:disable-line:max-line-length
const cls = (className, harFeil) => classNames(className, {
    'skjema__feilomrade--harFeil': harFeil
});
/**
 * Fieldset
 */
class SkjemaGruppe extends React.Component {
    renderTitle() {
        return (React.createElement("div", { className: "skjemaelement__sporsmal" }, this.props.title));
    }
    render() {
        const { children, className, title, feil, ...other } = this.props;
        return (React.createElement("div", Object.assign({ className: cls(className, feil) }, other),
            title && this.renderTitle(),
            children,
            React.createElement(skjemaelement_feilmelding_1.default, { feil: feil })));
    }
}
SkjemaGruppe.propTypes = {
    /**
     * En eller flere children, oftest en eller flere .skjemaelement
     */
    children: PT.oneOfType([
        PT.arrayOf(PT.node),
        PT.node
    ]).isRequired,
    /**
     * ClassName
     */
    className: PT.string,
    /**
     * Tittel eller skjemaspørsmål for skjemagruppe
     */
    title: PT.string,
    /**
     * Hvis skjemagruppen har feil sender man inn et objekt med en feilmelding
     */
    feil: skjemaelement_feilmelding_1.skjemaelementFeilmeldingShape
};
SkjemaGruppe.defaultProps = {
    className: undefined,
    feil: undefined,
    title: undefined
};
exports.default = SkjemaGruppe;
