// nav-frontend-skjema/src/skjemaelement-feilmelding.tsx
import * as React from 'react';
import * as PT from 'prop-types';

export type SkjemaelementFeilmeldingType = { feilmelding: string | React.ReactNode };

export interface SkjemaelementFeilmeldingProps {
    feil?: SkjemaelementFeilmeldingType;
}

class SkjemaelementFeilmelding extends React.Component<SkjemaelementFeilmeldingType> {

    renderFeil() {
        return (<div className="skjemaelement__feilmelding">{this.props.feil.feilmelding}</div>);
    }

    render() {
        const {feil} = this.props;
        return (
            <div role="alert" aria-live="assertive">
                {feil && this.renderFeil()}
            </div>
        );
    }
}

export const skjemaelementFeilmeldingShape = PT.shape({
    feilmelding: PT.oneOfType([
        PT.string,
        PT.node // For å slippe igjennom FormattedMessage fra react-intl
    ]).isRequired
});


(SkjemaelementFeilmelding as React.ComponentClass).propTypes = {
    /**
     * Hvis skjemaet har feil sender man inn et objekt med en feilmelding
     */
    feil: skjemaelementFeilmeldingShape
};

(SkjemaelementFeilmelding as React.ComponentClass).defaultProps = {
    feil: undefined
};

export default SkjemaelementFeilmelding;
