// nav-frontend-skjema/src/toggle-gruppe.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
const PT = require("prop-types");
require("nav-frontend-skjema-style");
const classNames = require("classnames");
class ToggleGruppe extends React.Component {
    render() {
        const { children, name, onChange, className } = this.props;
        const cls = (classes) => classNames('toggle', classes);
        return (React.createElement("div", { className: cls(className) }, children.map((toggleKnapp) => (React.cloneElement(toggleKnapp, { name, onChange, key: toggleKnapp.props.value })))));
    }
}
ToggleGruppe.defaultProps = {
    className: '',
    onChange: undefined
};
ToggleGruppe.propTypes = {
    /**
     * Et sett med ToggleKnapp-er
     */
    children: PT.node.isRequired,
    /**
     * className
     */
    className: PT.string,
    /**
     * Navnet som brukes på radio button gruppen
     */
    name: PT.string.isRequired,
    /**
     * Funksjon som håndterer endring av valgt knapp.
     * Funksjonen tar event. Iden til det klikkede elementet ligger på e.target.value
     * onChange kalles hver gang en knapp blir trykket.
     */
    onChange: PT.func
};
exports.default = ToggleGruppe;
