// nav-frontend-skjema/src/toggle-knapp.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
const PT = require("prop-types");
const nav_frontend_js_utils_1 = require("nav-frontend-js-utils");
require("nav-frontend-skjema-style");
const classNames = require("classnames");
class ToggleKnapp extends React.Component {
    render() {
        const { children, name, defaultChecked, value, onChange, checked, className } = this.props;
        if (checked !== undefined && !onChange) { // Trenger bare onChange om man definerer checked
            console.warn(`onChange har verdien: ${onChange}`);
        }
        const knappId = nav_frontend_js_utils_1.guid();
        const cls = (classes) => classNames('toggle__label', classes);
        return (React.createElement("div", null,
            React.createElement("input", { type: "radio", className: "toggle__input sr-only", defaultChecked: defaultChecked, onChange: onChange, name: name, id: knappId, value: value, checked: checked, "aria-checked": checked }),
            React.createElement("label", { className: cls(className), htmlFor: knappId }, children)));
    }
}
ToggleKnapp.defaultProps = {
    defaultChecked: undefined,
    checked: undefined,
    className: undefined,
    onChange: undefined,
    name: 'ToggleKnapp-default-name'
};
ToggleKnapp.propTypes = {
    /**
     * Innhold i label
     */
    children: PT.node.isRequired,
    /**
     * Navnet som brukes på radio knappen
     */
    name: PT.string,
    /**
     * Funksjon som håndterer endring av valgt knapp.
     * Funksjonen tar event. Iden til det klikkede elementet ligger på e.target.value
     * onChange kalles hver gang en knapp blir trykket.
     */
    onChange: PT.func,
    /**
     * Mulighet til å oppgi iden til en knapp som vil være satt som default
     */
    defaultChecked: PT.bool,
    /**
     * Value som brukes på alle ToggleKnapper i gruppen
     */
    value: PT.string.isRequired,
    /**
     * Marker en knapp som valgt
     */
    checked: PT.bool,
    /**
     * Classname
     */
    className: PT.string
};
exports.default = ToggleKnapp;
