// nav-frontend-skjema/src/toggle-gruppe.tsx
import * as React from 'react';
import * as PT from 'prop-types';
import * as classNames from 'classnames';
import 'nav-frontend-skjema-style'; // eslint-disable-line import/extensions

export interface ToggleGruppe {
    children: React.ReactNode | React.ReactChildren | React.ReactChild;
    name: string;
    className?: string;
    onChange?: () => void;
}

class ToggleGruppe extends React.Component { // eslint-disable-line react/prefer-stateless-function
    render() {
        const { children, name, onChange, className } = this.props;
        const cls = (classes) => classNames('toggle', classes);

        return (
            <div className={cls(className)}>
                { children.map((toggleKnapp) => (
                    React.cloneElement(toggleKnapp, { name, onChange, key: toggleKnapp.props.value })))
                }
            </div>
        );
    }
}

(ToggleGruppe as React.ComponentClass).defaultProps = {
    className: '',
    onChange: undefined
};

(ToggleGruppe as React.ComponentClass).propTypes = {
    /**
     * Et sett med ToggleKnapp-er
     */
    children: PT.node.isRequired,
    /**
     * className
     */
    className: PT.string,
    /**
     * Navnet som brukes på radio button gruppen
     */
    name: PT.string.isRequired,
    /**
     * Funksjon som håndterer endring av valgt knapp.
     * Funksjonen tar event. Iden til det klikkede elementet ligger på e.target.value
     * onChange kalles hver gang en knapp blir trykket.
     */
    onChange: PT.func
};

export default ToggleGruppe;
