// nav-frontend-tabs/src/tab.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
const cn = require("classnames");
const nav_frontend_js_utils_1 = require("nav-frontend-js-utils");
const tabCls = (props) => cn('nav-frontend-tabs__tab-inner', props.className, {
    'nav-frontend-tabs__tab-inner--aktiv': props.aktiv,
    'nav-frontend-tabs__tab-inner--interaktiv': !props.aktiv,
    'nav-frontend-tabs__tab-inner--disabled': props.disabled
});
class Tab extends React.Component {
    getLabel() {
        return (React.createElement("div", { className: "nav-frontend-tabs__tab-label" }, this.props.label || this.props.children));
    }
    render() {
        const { linkCreator = (props) => React.createElement("button", Object.assign({}, props)) } = this.props;
        const domProps = nav_frontend_js_utils_1.omit(this.props, 'label', 'aktiv', 'disabled', 'linkCreator', 'children', 'linkRef');
        return (React.createElement("li", { className: "nav-frontend-tabs__tab-outer", role: "presentation" }, linkCreator({
            ...domProps,
            className: tabCls(this.props),
            role: 'tab',
            'aria-selected': this.props.aktiv,
            children: this.getLabel(),
            ref: this.props.linkRef
        })));
    }
}
exports.default = Tab;
