// nav-frontend-tabs/src/tabs.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
const nav_frontend_js_utils_1 = require("nav-frontend-js-utils");
const tab_1 = require("./tab");
const tabs_pure_1 = require("./tabs-pure");
require("nav-frontend-tabs-style");
class Tabs extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            aktivTab: props.defaultAktiv
        };
    }
    setActiveTab(e, index) {
        this.setState({ aktivTab: index }, () => {
            this.pure.linkRefs[index].focus();
        });
        this.props.onChange(e, index);
    }
    renderTabs() {
        if (this.props.children) {
            return React.Children.map(this.props.children, (child, i) => {
                if (React.isValidElement(child)) {
                    return React.cloneElement(child, {
                        aktiv: (i === this.state.aktivTab)
                    });
                }
                return child;
            });
        }
        if (this.props.tabs) {
            return this.props.tabs.map((tab, i) => {
                return (React.createElement(tab_1.default, Object.assign({ key: nav_frontend_js_utils_1.guid(), aktiv: i === this.state.aktivTab }, tab)));
            });
        }
    }
    render() {
        const domProps = nav_frontend_js_utils_1.omit(this.props, 'tabs', 'onChange', 'defaultAktiv', 'ref');
        return (React.createElement(tabs_pure_1.default, Object.assign({ ref: (pure) => this.pure = pure, onChange: (e, index) => this.setActiveTab(e, index) }, domProps), this.renderTabs()));
    }
}
Tabs.Tab = tab_1.default;
Tabs.defaultProps = {
    kompakt: false,
    defaultAktiv: 0
};
exports.default = Tabs;
var tabs_pure_2 = require("./tabs-pure");
exports.TabsPure = tabs_pure_2.default;
