// nav-frontend-grid/src/column.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const PT = require("prop-types");
const React = require("react");
const classNames = require("classnames");
require("nav-frontend-grid-style");
const cls = (className, xs, sm, md, lg) => classNames('col', className, {
    [`col-xs-${xs}`]: !!xs,
    [`col-sm-${sm}`]: !!sm,
    [`col-md-${md}`]: !!md,
    [`col-lg-${lg}`]: !!lg
});
class Column extends React.Component {
    render() {
        const { children, className, xs, sm, md, lg, ...props } = this.props;
        return (React.createElement("div", Object.assign({ className: cls(className, xs, sm, md, lg) }, props), children));
    }
}
Column.defaultProps = {
    className: undefined,
    children: undefined,
    xs: undefined,
    sm: undefined,
    md: undefined,
    lg: undefined
};
Column.propTypes = {
    className: PT.string,
    xs: PT.oneOf(['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12']),
    sm: PT.oneOf(['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12']),
    md: PT.oneOf(['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12']),
    lg: PT.oneOf(['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12']),
    children: PT.oneOfType([
        PT.arrayOf(PT.node),
        PT.node
    ])
};
exports.default = Column;
