// nav-frontend-lenkepanel/src/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const PT = require("prop-types");
const React = require("react");
const nav_frontend_typografi_1 = require("nav-frontend-typografi");
require("nav-frontend-lenkepanel-style");
const Lenkepanel_base_1 = require("./Lenkepanel-base");
class Lenkepanel extends React.PureComponent {
    render() {
        const { children, tittelProps = 'undertittel', ...renderProps } = this.props;
        const headingConfig = { type: tittelProps, tag: 'span' };
        const heading = React.createElement(nav_frontend_typografi_1.default, Object.assign({ key: "heading" }, headingConfig, { className: "lenkepanel__heading" }), children);
        return (React.createElement(Lenkepanel_base_1.default, Object.assign({}, renderProps), heading));
    }
}
Lenkepanel.propTypes = {
    /**
     * Klassenavn
     */
    className: PT.string,
    /**
     * Lenke (denne sendes som prop til linkCreator)
     */
    href: PT.string.isRequired,
    /**
     * Tekstinnhold for lenkepanel
     */
    children: PT.oneOfType([
        PT.arrayOf(PT.node),
        PT.node
    ]).isRequired,
    /**
     * Typografitype
     */
    tittelProps: PT.string,
    /**
     * Funksjon som brukes for å lage lenken. her kan du spessifisere din egen hvis du f.eks. bruker React Router
     */
    linkCreator: PT.func
};
Lenkepanel.defaultProps = {
    className: undefined,
    tittelProps: 'undertittel',
    linkCreator: (props) => React.createElement("a", Object.assign({}, props))
};
exports.default = Lenkepanel;
var Lenkepanel_base_2 = require("./Lenkepanel-base");
exports.LenkepanelBase = Lenkepanel_base_2.default;
