// nav-frontend-chevron/src/chevron.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const PT = require("prop-types");
const React = require("react");
const classNames = require("classnames");
require("nav-frontend-chevron-style");
// export type ChevronType = 'høyre' | 'venstre' | 'ned' | 'opp';
const cls = (type, stor, className) => classNames('nav-frontend-chevron chevronboks', className, {
    'chevron--stor': stor
}, {
    'chevron--hoyre': type === 'høyre',
    'chevron--venstre': type === 'venstre',
    'chevron--ned': type === 'ned',
    'chevron--opp': type === 'opp'
});
class NavFrontendChevron extends React.Component {
    render() {
        const { type, stor, className, ...props } = this.props;
        return React.createElement("i", Object.assign({ className: cls(type, stor, className) }, props));
    }
}
NavFrontendChevron.defaultProps = {
    type: 'høyre',
    stor: false,
    className: ''
};
NavFrontendChevron.propTypes = {
    /**
     * Bestemmer hvilken vei Chevron skal peke
     */
    type: PT.oneOf(['høyre', 'venstre', 'opp', 'ned']),
    /**
     * Sett denne proppen hvis chevron skal være stor
     */
    stor: PT.bool,
    /**
     * Klassenavn for chevron
     */
    className: PT.string
};
exports.default = NavFrontendChevron;
var hoyre_chevron_1 = require("./hoyre-chevron");
exports.HoyreChevron = hoyre_chevron_1.default;
var venstre_chevron_1 = require("./venstre-chevron");
exports.VenstreChevron = venstre_chevron_1.default;
var opp_chevron_1 = require("./opp-chevron");
exports.OppChevron = opp_chevron_1.default;
var ned_chevron_1 = require("./ned-chevron");
exports.NedChevron = ned_chevron_1.default;
