// nav-frontend-veilederpanel/src/veilederpanel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
const PT = require("prop-types");
const cn = require("classnames");
const nav_frontend_veileder_1 = require("nav-frontend-veileder");
require("nav-frontend-veilederpanel-style");
const veilederPanelCls = (props) => cn('nav-veilederpanel', {
    'nav-veilederpanel--kompakt': props.kompakt,
    'nav-veilederpanel--plakat': props.type === 'plakat',
    'nav-veilederpanel--suksess': props.fargetema === 'suksess',
    'nav-veilederpanel--advarsel': props.fargetema === 'advarsel',
    'nav-veilederpanel--feilmelding': props.fargetema === 'feilmelding'
});
class Veilederpanel extends React.Component {
    render() {
        const { svg, children } = this.props;
        const storrelse = this.props.type === 'plakat' ? 'M' : 'S';
        return (React.createElement("div", { className: veilederPanelCls(this.props) },
            React.createElement(nav_frontend_veileder_1.default, Object.assign({}, this.props.veilederProps, { fargetema: this.props.fargetema, storrelse: storrelse }), svg),
            React.createElement("div", null, children)));
    }
}
Veilederpanel.propTypes = {
    /**
     * Egendefinert SVG som rendres i en Veileder-boble
     */
    svg: PT.node.isRequired,
    /**
     * Predefinerte fargetemaer som bestemmer bakgrunnsfargen på veilederen og
     * border på veilederpanelet.
     */
    fargetema: PT.oneOf(['normal', 'suksess', 'advarsel', 'feilmelding']),
    /**
     * Verdiene 'normal' og 'plakat' bestemmer henholdsvis om veilederen plasseres
     * til venstre eller over innholdet.
     */
    type: PT.oneOf(['normal', 'plakat']),
    /**
     * Kompakt versjon av veilederpanelet bruker litt mindre plass enn normalt, hvor
     * veilderen rendres overlappende med border på panelet.
     */
    kompakt: PT.bool,
    /**
     * Props som blir videreformidlet til intern instans av Veileder.
     */
    veilederProps: nav_frontend_veileder_1.VeilederPropsShape
};
exports.default = Veilederpanel;
