// nav-frontend-snakkeboble/src/snakkeboble.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
const classNames = require("classnames");
const prop_types_1 = require("prop-types");
require("nav-frontend-snakkeboble-style");
const nav_frontend_paneler_1 = require("nav-frontend-paneler");
const nav_frontend_typografi_1 = require("nav-frontend-typografi");
const snakkebobleCls = (hoyre) => classNames('snakkeboble', {
    'snakkeboble--hoyre': hoyre, 'snakkeboble--venstre': !hoyre
});
/**
 * En snakkeboble for dialog
 */
class Snakkeboble extends React.Component {
    render() {
        const { dato, topp, children, pilHoyre, ikonClass } = this.props;
        const toppTekst = topp || dato;
        return (React.createElement("div", { className: snakkebobleCls(pilHoyre) },
            React.createElement("i", { className: ikonClass }),
            React.createElement("div", { className: "snakkeboble__snakkeboble-pil-container" },
                React.createElement("i", { className: "snakkeboble__snakkebole-pil" })),
            React.createElement(nav_frontend_paneler_1.Panel, { className: "snakkeboble-panel" },
                toppTekst &&
                    React.createElement(nav_frontend_typografi_1.Undertekst, { className: "snakkeboble-panel__topp" }, toppTekst),
                React.createElement("div", { className: "snakkeboble-panel__tekst" }, children))));
    }
}
Snakkeboble.defaultProps = {
    /**
     * Default er at pilen peker mot venstre
     */
    pilHoyre: false,
    /**
     * Gir enkel default styling på ikoner
     */
    ikonClass: 'snakkeboble__ikon',
    /**
     * @Deprecated Bruk `topp`-prop
     */
    dato: null
};
Snakkeboble.propTypes = {
    /**
     * Toppteksten (tittel) for meldingen
     */
    topp: prop_types_1.PropTypes.string,
    /**
     * @Deprecated Dato for melding, bruk `topp`
     */
    dato: prop_types_1.PropTypes.string,
    /**
     * Tekst for melding
     */
    children: prop_types_1.PropTypes.node.isRequired,
    /**
     * Settes denne peker pilen til snakkeboblen mot høyre, og snakkeboblen og teksten i snakkeboblen flyter mot høyre.
     */
    pilHoyre: prop_types_1.PropTypes.bool,
    /**
     * Klasse som brukes på ikonet. Dette lar deg styre form, farge og illustrasjon.
     */
    ikonClass: prop_types_1.PropTypes.string
};
exports.default = Snakkeboble;
