// nav-frontend-ekspanderbartpanel/src/ekspanderbartpanel-base.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
require("nav-frontend-ekspanderbartpanel-style");
const ekspanderbartpanel_base_pure_1 = require("./ekspanderbartpanel-base-pure");
class EkspanderbartpanelBase extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            apen: this.props.apen
        };
        this.handleClick = this.handleClick.bind(this);
    }
    handleClick(event) {
        event.preventDefault();
        this.setState({ apen: !this.state.apen });
        if (this.props.onClick) {
            this.props.onClick(event);
        }
    }
    render() {
        const { ...renderProps } = this.props;
        return (React.createElement(ekspanderbartpanel_base_pure_1.default, Object.assign({}, renderProps, { apen: this.state.apen, onClick: this.handleClick })));
    }
}
EkspanderbartpanelBase.defaultProps = {
    apen: false,
    border: false
};
exports.default = EkspanderbartpanelBase;
