// nav-frontend-ekspanderbartpanel/src/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
require("nav-frontend-ekspanderbartpanel-style");
const ekspanderbartpanel_pure_1 = require("./ekspanderbartpanel-pure");
class Ekspanderbartpanel extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            apen: this.props.apen
        };
        this.handleClick = this.handleClick.bind(this);
    }
    handleClick(event) {
        event.preventDefault();
        this.setState({ apen: !this.state.apen });
        if (this.props.onClick) {
            this.props.onClick(event);
        }
    }
    render() {
        const { tittel, tittelProps, ...renderProps } = this.props;
        return (React.createElement(ekspanderbartpanel_pure_1.default, Object.assign({}, renderProps, { apen: this.state.apen, onClick: this.handleClick, tittel: tittel, tittelProps: tittelProps })));
    }
}
Ekspanderbartpanel.defaultProps = {
    apen: false,
    border: false
};
exports.default = Ekspanderbartpanel;
var ekspanderbartpanel_pure_2 = require("./ekspanderbartpanel-pure");
exports.EkspanderbartpanelPure = ekspanderbartpanel_pure_2.default;
var ekspanderbartpanel_base_1 = require("./ekspanderbartpanel-base");
exports.EkspanderbartpanelBase = ekspanderbartpanel_base_1.default;
var ekspanderbartpanel_base_pure_1 = require("./ekspanderbartpanel-base-pure");
exports.EkspanderbartpanelBasePure = ekspanderbartpanel_base_pure_1.default;
