// nav-frontend-veileder/src/veileder-chat-bubble.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
const PT = require("prop-types");
class VeilederChatBubble extends React.Component {
    render() {
        const { says, offset } = this.props;
        const scaleMultiplier = Math.max(0.5, offset / 150);
        const bubbleStyles = {
            bottom: offset + ((20 * scaleMultiplier) / 2),
            marginRight: offset / 2
        };
        const arrowStyles = {
            bottom: Math.ceil(-20 - (((20 * scaleMultiplier) / 2) - 10)),
            right: ((20 - (20 * scaleMultiplier)) * -1) + (20 * scaleMultiplier),
            transform: `scale(${scaleMultiplier})`
        };
        return (React.createElement("div", { className: "nav-frontend-veileder__chatBubble", style: bubbleStyles },
            says,
            React.createElement("i", { className: "nav-frontend-veileder__chatBubbleArrow", style: arrowStyles })));
    }
}
VeilederChatBubble.propTypes = {
    says: PT.node.isRequired,
    offset: PT.number.isRequired
};
exports.default = VeilederChatBubble;
