// nav-frontend-veileder/src/veileder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
const PT = require("prop-types");
const cn = require("classnames");
const nav_frontend_js_utils_1 = require("nav-frontend-js-utils");
require("nav-frontend-veileder-style");
const cls = (props) => cn('nav-veileder', {
    'nav-veileder--advarsel': props.type === 'advarsel',
    'nav-veileder--feilmelding': props.type === 'feilmelding',
    'nav-veileder--suksess': props.type === 'suksess'
});
const frameCls = (props) => cn('nav-veileder__frame', {
    'nav-veileder__frame--center': props.center,
    'nav-veileder__frame--transparent': props.transparent,
    'nav-veileder__frame--nomask': props.nomask,
    'nav-veileder__frame--s': props.storrelse === 'S',
    'nav-veileder__frame--m': props.storrelse === 'M',
    'nav-veileder__frame--l': props.storrelse === 'L',
    'nav-veileder__frame--xl': props.storrelse === 'XL',
    'nav-veileder__frame--suksess': props.fargetema === 'suksess',
    'nav-veileder__frame--advarsel': props.fargetema === 'advarsel',
    'nav-veileder__frame--feilmelding': props.fargetema === 'feilmelding'
});
class Veileder extends React.Component {
    constructor() {
        super(...arguments);
        this.renderSnakkeboble = () => {
            return (React.createElement("div", { className: "nav-veileder__snakkeboble" },
                this.props.tekst,
                React.createElement("i", { className: "nav-veileder__snakkeboblePil" })));
        };
    }
    render() {
        const { children, tekst, ...other } = this.props;
        const domProps = nav_frontend_js_utils_1.omit({ ...other }, 'center', 'nomask', 'says', 'color', 'transparent', 'width', 'type', 'fargetema', 'storrelse');
        return (React.createElement("div", Object.assign({ className: cls(this.props) }, domProps),
            React.createElement("div", { className: frameCls(this.props) }, children),
            tekst && this.renderSnakkeboble()));
    }
}
Veileder.defaultProps = {
    type: 'normal',
    fargetema: 'normal',
    storrelse: 'M'
};
exports.VeilederPropsShape = PT.shape({
    /**
     * Custom SVG-element som illustrerer en figur, f.eks. en av disse:
     * https://app.zeplin.io/project/59831e455850985791bdeb4d/screen/5a7c1fcc36781f9423ab6a0e
     */
    children: PT.node.isRequired,
    /**
     * Tekst eller HTML som rendres i en snakkeboble
     */
    tekst: PT.node,
    /**
     * Skrur av bakgrunnsfargen
     */
    transparent: PT.bool,
    /**
     * Skrur av sirkulær maskering/cropping av children
     */
    nomask: PT.bool,
    /**
     * Skrur av 80% height på children, og rendrer i full høyde (hvis innholdet skal sentreres)
     */
    center: PT.bool,
    /**
     * Predefinerte nøkler for styling av snakkeboblen, avhengig av type budskap.
     */
    type: PT.oneOf(['normal', 'suksess', 'advarsel', 'feilmelding']),
    /**
     * Predefinerte fargetemaer som bestemmer bakgrunnsfargen på veilederen.
     */
    fargetema: PT.oneOf(['normal', 'suksess', 'advarsel', 'feilmelding']),
    /**
     * Predefinerte størrelser
     */
    storrelse: PT.oneOf(['S', 'M', 'L', 'XL'])
});
Veileder.propTypes = exports.VeilederPropsShape;
Veileder.defaultProps = {
    type: 'normal',
    fargetema: 'normal',
    storrelse: 'M'
};
exports.default = Veileder;
