// nav-frontend-typografi/src/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const PT = require("prop-types");
const React = require("react");
const classNames = require("classnames");
require("nav-frontend-typografi-style");
exports.typeConfigMap = {
    sidetittel: { defaultTag: 'h1', cls: (...args) => classNames('typo-sidetittel', ...args) },
    innholdstittel: { defaultTag: 'h1', cls: (...args) => classNames('typo-innholdstittel', ...args) },
    systemtittel: { defaultTag: 'h2', cls: (...args) => classNames('typo-systemtittel', ...args) },
    undertittel: { defaultTag: 'h2', cls: (...args) => classNames('typo-undertittel', ...args) },
    ingress: { defaultTag: 'p', cls: (...args) => classNames('typo-ingress', ...args) },
    element: { defaultTag: 'p', cls: (...args) => classNames('typo-element', ...args) },
    normaltekst: { defaultTag: 'p', cls: (...args) => classNames('typo-normal', ...args) },
    etikettLiten: { defaultTag: 'p', cls: (...args) => classNames('typo-etikett-liten', ...args) },
    undertekstBold: { defaultTag: 'p', cls: (...args) => classNames('typo-undertekst-bold', ...args) },
    undertekst: { defaultTag: 'p', cls: (...args) => classNames('typo-undertekst', ...args) }
};
function getConfigForType(type) {
    const typeConfig = exports.typeConfigMap[type];
    if (!typeConfig) {
        throw new Error(`Kunne ikke finne typeconfig for ${type}. Støttede typer er: ${JSON.stringify(Object.keys(exports.typeConfigMap))}`);
    }
    return typeConfig;
}
class TypografiBase extends React.Component {
    render() {
        const { type, tag, className, children, ...props } = this.props;
        const config = getConfigForType(type);
        const tagName = tag || config.defaultTag;
        const classnames = { className: config.cls(className) };
        return React.createElement(tagName, { ...props, ...classnames }, children);
    }
}
TypografiBase.propTypes = {
    type: PT.string.isRequired,
    tag: PT.string,
    className: PT.string,
    children: PT.node.isRequired
};
TypografiBase.defaultProps = {
    className: undefined
};
exports.default = TypografiBase;
var sidetittel_1 = require("./sidetittel");
exports.Sidetittel = sidetittel_1.default;
var innholdstittel_1 = require("./innholdstittel");
exports.Innholdstittel = innholdstittel_1.default;
var systemtittel_1 = require("./systemtittel");
exports.Systemtittel = systemtittel_1.default;
var undertittel_1 = require("./undertittel");
exports.Undertittel = undertittel_1.default;
var ingress_1 = require("./ingress");
exports.Ingress = ingress_1.default;
var element_1 = require("./element");
exports.Element = element_1.default;
var normaltekst_1 = require("./normaltekst");
exports.Normaltekst = normaltekst_1.default;
var etikett_liten_1 = require("./etikett-liten");
exports.EtikettLiten = etikett_liten_1.default;
var undertekst_1 = require("./undertekst");
exports.Undertekst = undertekst_1.default;
var undertekst_bold_1 = require("./undertekst-bold");
exports.UndertekstBold = undertekst_bold_1.default;
