// nav-frontend-alertstriper/src/alertstripe.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const PT = require("prop-types");
const React = require("react");
const classNames = require("classnames");
const nav_frontend_typografi_1 = require("nav-frontend-typografi");
const nav_frontend_ikoner_assets_1 = require("nav-frontend-ikoner-assets");
require("nav-frontend-alertstriper-style");
const cls = (type, solid, className) => classNames('alertstripe', className, {
    'alertstripe--solid': solid && !(type === 'stopp'),
    'alertstripe--advarsel': type === 'advarsel' || type === 'nav-ansatt' || type === 'stopp',
    'alertstripe--info': type === 'info',
    'alertstripe--suksess': type === 'suksess'
});
const ikonKind = (type, solid) => {
    const kindMap = {
        'ok-sirkel-fylt': type === 'suksess' && solid,
        'ok-sirkel': type === 'suksess' && !solid,
        'advarsel-trekant': type === 'advarsel' && !solid,
        'advarsel-trekant-fylt': type === 'advarsel' && solid,
        'info-sirkel-fylt': type === 'info' && solid,
        'info-sirkel-orange': type === 'info' && !solid,
        'nav-ansatt': type === 'nav-ansatt',
        'feil-sirkel-fylt': type === 'stopp'
    };
    return Object.keys(kindMap)
        .map((key) => ({ key, predicate: kindMap[key] }))
        .find(({ predicate }) => predicate).key;
};
class AlertStripe extends React.Component {
    render() {
        const { type, solid, size, className, children, ...props } = this.props;
        const ariaLive = (type === 'advarsel') ? 'assertive' : undefined;
        return (React.createElement("div", Object.assign({ className: cls(type, solid, className) }, props, { "aria-live": ariaLive }),
            React.createElement("span", { className: "alertstripe__ikon", "aria-label": type },
                React.createElement(nav_frontend_ikoner_assets_1.default, { kind: ikonKind(type, solid), size: size })),
            React.createElement(nav_frontend_typografi_1.Normaltekst, { className: "alertstripe__tekst", tag: "span" }, children)));
    }
}
AlertStripe.propTypes = {
    children: PT.node.isRequired,
    type: PT.oneOf(['suksess', 'info', 'advarsel', 'nav-ansatt', 'stopp']).isRequired,
    solid: PT.bool,
    size: PT.oneOfType([PT.string, PT.number]),
    className: PT.string
};
AlertStripe.defaultProps = {
    className: undefined,
    size: '1.5em',
    solid: false
};
exports.default = AlertStripe;
var advarsel_solid_alertstripe_1 = require("./advarsel-solid-alertstripe");
exports.AlertStripeAdvarselSolid = advarsel_solid_alertstripe_1.default;
var advarsel_alertstripe_1 = require("./advarsel-alertstripe");
exports.AlertStripeAdvarsel = advarsel_alertstripe_1.default;
var suksess_solid_alertstripe_1 = require("./suksess-solid-alertstripe");
exports.AlertStripeSuksessSolid = suksess_solid_alertstripe_1.default;
var suksess_alertstripe_1 = require("./suksess-alertstripe");
exports.AlertStripeSuksess = suksess_alertstripe_1.default;
var info_solid_alertstripe_1 = require("./info-solid-alertstripe");
exports.AlertStripeInfoSolid = info_solid_alertstripe_1.default;
var info_alertstripe_1 = require("./info-alertstripe");
exports.AlertStripeInfo = info_alertstripe_1.default;
var navansatt_alertstripe_1 = require("./navansatt-alertstripe");
exports.AlertStripeNavAnsatt = navansatt_alertstripe_1.default;
var stopp_alertstripe_1 = require("./stopp-alertstripe");
exports.AlertStripeStopp = stopp_alertstripe_1.default;
